backend/src/main/java/de/laura/project/db/TempTriviaQuestionDB.java

```diff
@@ -17,38 +17,48 @@
 @NoArgsConstructor
 @Builder
 
-@Service
+@Repository
 public class TempTriviaQuestionDB {
 
-    private List<TriviaQuestionSet> triviaQuestionSetList;
+    private List<TriviaQuestionSet> triviaQuestionSetList = new ArrayList<>();
 
 
     public List<TriviaQuestionSet> returnQuestionList(List<TriviaQuestionSet> triviaQuestionSetList) {
-        this.triviaQuestionSetList = triviaQuestionSetList;
         return triviaQuestionSetList;
     }
 
-    public TriviaQuestionSetWithoutAnswer getSingleQuestion(int id) {
+    public TriviaQuestionSetWithoutCorrectAnswer getSingleQuestion(int questionId) {
 
-        List<TriviaQuestionSetWithoutAnswer> triviaQuestionSetListWithoutAnswer = new ArrayList<>();
+        List<TriviaQuestionSetWithoutCorrectAnswer> triviaQuestionSetListWithoutAnswer = new ArrayList<>();
 
         triviaQuestionSetListWithoutAnswer = triviaQuestionSetList.stream()
-                .map(triviaQuestionSet -> TriviaQuestionSetWithoutAnswer.builder()
+                .map(triviaQuestionSet -> TriviaQuestionSetWithoutCorrectAnswer.builder()
                         .id(triviaQuestionSet.getId())
                         .question(triviaQuestionSet.getQuestion())
                         .answers(triviaQuestionSet.getAnswers())
                         .build()).collect(Collectors.toList());
 
+        return triviaQuestionSetListWithoutAnswer.stream()
+                .filter(triviaQuestionSet -> triviaQuestionSet.getId() == (questionId))
+                .findFirst().get();
```

Comment:
```suggestion
                .findAny().get().orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "ID could not be found"));
```
if the order in your DB is irrelevant you can also use `.findAny()`

https://www.baeldung.com/java-stream-findfirst-vs-findany